openedx/core/djangoapps/user_authn/views/password_reset.py

```diff
@@ -299,171 +300,226 @@ def _uidb36_to_uidb64(uidb36):
     return uidb64
 
 
-# pylint: disable=too-many-statements
-def password_reset_confirm_wrapper(request, uidb36=None, token=None):
+class PasswordResetConfirmWrapper(PasswordResetConfirmView):
     """
-    A wrapper around django.contrib.auth.views.password_reset_confirm.
-    Needed because we want to set the user as active at this step.
-    We also optionally do some additional password policy checks.
+    A wrapper around django.contrib.auth.views.PasswordResetConfirmView.
+      Needed because we want to set the user as active at this step.
+      We also optionally do some additional password policy checks.
     """
-    # convert old-style base36-encoded user id to base64
-    uidb64 = _uidb36_to_uidb64(uidb36)
-    platform_name = {
-        "platform_name": configuration_helpers.get_value('platform_name', settings.PLATFORM_NAME)
-    }
 
-    # User can not get this link unless account recovery feature is enabled.
-    if 'is_account_recovery' in request.GET and not is_secondary_email_feature_enabled():
-        raise Http404
+    reset_url_token = 'set-password'
+
+    def __init__(self):
+        self.post_reset_login = False
+        self.platform_name = PasswordResetConfirmWrapper._get_platform_name()
+        self.validlink = False
+        self.user = None
+        self.uidb36 = ''
+        self.token = ''
+        self.uidb64 = ''
+        self.uid_int = -1
+
+    def _process_password_reset_success(self, request, token, uidb64, extra_context):
+        self.user = self.get_user(uidb64)
+        form = SetPasswordForm(self.user, request.POST)
+        if self.token_generator.check_token(self.user, token) and form.is_valid():
+            self.form_valid(form)
```

Comment:
@mraarif any idea why we need `self.form_valid(form)` here when we already checked the form validation in the `if condition` above?